### DISCUSSION TITLE
[v6] Standalone <Route>

### DISCUSSION THREAD

**Discussion by mjackson** [remix-run/react-router#8018] [Proposals]
### What is the new or updated feature that you are suggesting?

We should make it easy to render a standalone `<Route>` without requiring it to be wrapped in a `<Routes>`. This mirrors functionality we currently have in v5.

Unfortunately this probably means we will have to get rid of the *secret menu* in v6, but that's hard to type anyway. Reach Router users will have to modify their route config to use real `<Route>` components instead of sending `<Route>`-specific props to their own components.

### Why should this feature be included?

This will make it easier for people to upgrade from v5.

**Comment by timdorr** [comment]
Maybe it makes sense to bring back the old `<Match>` component? 🤔

**Comment by MeiKatz** [comment]
Just for those who don't know it: what is the secret menu in v6?

**Comment by timdorr** [comment]
That's where you can get React Router in Animal Style, Protein Style, or get it 4x4.

**Comment by mjackson** [comment]
@MeiKatz https://github.com/remix-run/react-router/blob/bf679c49d8f6192640948adb8a44d60c8768f0e3/packages/react-router/__tests__/route-matching-test.tsx#L75

**Comment by MeiKatz** [comment]
So the secret menu is adding `path` to any kind of component, that is not a `<Route />`?

**Comment by mjackson** [comment]
It's essentially the Reach Router API for React Router v6, where you can use any component you want instead of a `<Route>`. You just give it the same props that `<Route>` accepts.

It's problematic when you're using TypeScript, because now you have to add `<Route>`'s props to the props that all of your components expect, so I wouldn't be terribly sad to lose it.

**Comment by mjackson** [comment]
We discussed this internally and decided it's probably not the direction we want to go right now.

Instead of implementing standalone `<Route>` in v6, we are going to ship a backwards compatibility layer that will implement the v5 API on top of v6 (including a standalone `<Route>` component), which should help people when migrating.
